### PR TITLE
Add MODIS 05 L2 datasets to `modis_l2` reader

### PR DESCRIPTION
**Pull request by bornagain1981**
Hi guys, below a couple of product modis05 datasets .  content of water vapor with retrieval in infrared and near infrared bands, correction factor , vza, ad acquisition time datasets (for mod06 and mod05 both)
I think the tests are enough for those that already exist

### CODE REVIEW

**Comment by bornagain1981** [comment]
not accepted 2 file_key having same name: "Scan_Start_Time"
This is the name of variable in hdf, correct?
So this is the same name in MOD05 and MOD06, how to solve it?

**Comment by simonrp84** [comment]
I think you need to delete that key from the MOD05 section. Then, in the MOD06 section, change file type to:
`file_type: [mod05_hdf, mod06_hdf]`

File: satpy/etc/readers/modis_l2.yaml [hunk 1]
```diff
@@ -143,12 +148,75 @@ datasets:
...
+    units: seconds since 1993-1-1 00:00:00.0 0
+    file_type: mod05_hdf
+    coordinates: [longitude, latitude]
+    resolution:
+      5000:
+        file_key: Scan_Start_Time
+
+  sensor_zenith:
+    name: sensor_zenith
+    long_name: Sensor Zenith Angle, Cell to Sensor
```
Comment (djhoese) [inline satpy/etc/readers/modis_l2.yaml:166]: For consistency with the rest of Satpy, what do people think about naming this `satellite_zenith_angle`?

**Review by djhoese — CHANGES_REQUESTED** [review]
Looks pretty good. One small name change request for the sensor angle. Otherwise, I agree with Simon that you need to combine the scan_start_time using the `file_type`.

File: satpy/etc/readers/modis_l2.yaml (hunk 1 above)
Comment (simonrp84) [reply satpy/etc/readers/modis_l2.yaml:166]: Yes, good idea.

Comment (bornagain1981) [reply satpy/etc/readers/modis_l2.yaml]: done

File: satpy/etc/readers/modis_l2.yaml [hunk 2]
```diff
@@ -143,6 +148,59 @@ datasets:
...
+    file_type: [mod05_hdf, mod06_hdf]
+    coordinates: [longitude, latitude]
+    resolution:
+      5000:
+        file_key: Sensor_Zenith
+
+  water_vapor_correction_factor:
+    name: water_vapor_correction_factor
+    long_name: Aerosol Correction Factor for Water Vapor - Near Infrared Retrieval
+    units: None
```
[the remark this answers is not in the document]
Comment (bornagain1981) [reply satpy/etc/readers/modis_l2.yaml:177]: using Panoply  shows this
 :units = "none" 

`short Water_Vapor_Correction_Factors(Cell_Along_Swath_1km=2030, Cell_Across_Swath_1km=1354);
  :long_name = "Aerosol Correction Factor for Water Vapor - Near Infrared Retrieval";
  :add_offset = 0.0; // double
  :Parameter_Type = "Output";
  :Cell_Along_Swath_Sampling = 1, 2030, 1; // int
  :Cell_Across_Swath_Sampling = 1, 1354, 1; // int
  :Geolocation_Pointer = "Internal geolocation arrays";
  :valid_range = 0S, 20000S; // short
  :_FillValue = -9999S; // short
  :units = "none";
  :scale_factor = 0.0010000000474974513; // double`

[the remark this answers is not in the document]
Comment (djhoese) [reply satpy/etc/readers/modis_l2.yaml:177]: 1. `"none"` != `None`
2. I wouldn't look to MODIS data as a source for CF standard metadata. I'm not sure CF was even used/existed when HDF-EOS/MODIS files were first created.
3. We're trying to be consistent with the rest of Satpy, not necessarily with MODIS L2 processing.
4. If valid units are in the data file we (the python code) should be pulling that metadata from the file, not configuring it in the YAML. In this case I don't think the units are valid though.

**Review by djhoese — APPROVED** [review]

**Pull request merged**